src/commands/reword.rs

```diff
@@ -0,0 +1,358 @@
+                return Ok(1);
+            }
+        }
+    };
+
+    let now = SystemTime::now();
+    let event_tx_id = event_log_db.make_transaction_id(now, "reword")?;
+    let execute_options = ExecuteRebasePlanOptions {
+        now,
+        event_tx_id,
+        preserve_timestamps: get_restack_preserve_timestamps(&repo)?,
+        force_in_memory: true,
+        force_on_disk: false,
+        resolve_merge_conflicts: false,
+        check_out_commit_options: CheckOutCommitOptions {
+            additional_args: Default::default(),
+            render_smartlog: false,
+        },
+    };
+    let result = execute_rebase_plan(effects, git_run_info, &repo, &rebase_plan, &execute_options)?;
+
+    let exit_code = match result {
+        ExecuteRebasePlanResult::Succeeded { rewritten_oids } => {
+            render_status_report(&repo, effects, &commits, &rewritten_oids)?;
+
+            0
+        }
+        ExecuteRebasePlanResult::DeclinedToMerge { merge_conflict: _ } => {
+            writeln!(
+                effects.get_error_stream(),
+                "BUG: Merge conflict detected, but rewording shouldn't cause any conflicts."
+            )?;
+
+            1
+        }
+        ExecuteRebasePlanResult::Failed { exit_code } => exit_code,
+    };
+
+    Ok(exit_code)
+}
+
+/// Turn a list of ref-ish strings into a list of Commits.
+fn resolve_commits_from_hashes<'repo>(
+    repo: &'repo Repo,
+    dag: &mut Dag,
+    effects: &Effects,
+    hashes: Vec<String>,
+) -> eyre::Result<Option<Vec<Commit<'repo>>>> {
+    let hashes = if hashes.is_empty() {
+        vec!["HEAD".to_string()]
+    } else {
+        hashes
+    };
+
+    let commits = resolve_commits(effects, repo, dag, hashes)?;
+    let commits = match commits {
+        ResolveCommitsResult::Ok { commits } => commits,
+        ResolveCommitsResult::CommitNotFound { commit: hash } => {
+            writeln!(effects.get_output_stream(), "Commit not found: {}", hash)?;
+            return Ok(None);
+        }
+    };
+    Ok(Some(commits))
+}
+
+/// The result of building the reword message.
+#[must_use]
+pub enum BuildRewordMessageResult {
+    /// The reworded message was built successfully.
+    Succeeded {
+        /// The reworded messages for each commit.
+        messages: HashMap<NonZeroOid, String>,
+    },
+
+    /// The reworded message matches the original message.
+    IdenticalMessage,
+
+    /// The reworded message was empty.
+    EmptyMessage,
+}
+
+/// Builds the message(s) that will be used for rewording. These are mapped from each commit's
+/// NonZeroOid to the relevant message.
+#[instrument]
+fn build_messages(
+    messages: &[String],
+    commits: &[Commit],
+    comment_char: Option<u8>,
+) -> eyre::Result<BuildRewordMessageResult> {
+    let message = messages.join("\n\n").trim().to_string();
+
+    let (message, load_editor) = if message.is_empty() {
+        let message_for_editor = match commits.to_vec().as_slice() {
```

Comment:
Oops, `commits` was already a slice, so we didn't need to call `as_slice()` on it at all.

```suggestion
        let message_for_editor = match commits {
```

Reply:
Ah, of course!